### ISSUE TITLE
Selectable theme for projects (sub sites)

### ISSUE THREAD

**Issue by PrinsMild** [Puzzlepart/prosjektportalen#423]
#### Category

[X] Enhancement

[ ] Bug

[ ] Question

#### Environment

[X] SharePoint Online

[X] SharePoint 2013

[X] SharePoint 2016

#### Summary

Projects / sites are set to green theme by default. This is enoying for those who want a different theme or layout, forceing heavy CSS overrides or manually changings each time a project is created.. 

#### Version

All versions

#### Expected / Desired Behavior / Question
Selecting themes should be an option in the ProjectConfig list or equal.

#### Steps to Reproduce
Create anew project (sub site)

**Comment by olemp** [comment]
Hi PrinsMild,

We have support for this. Upload a extension (.txt) file to /Extensions.

If you've installed PP in norwegian, you'll find `Prosjektportalen - Utvidelser` under `Utseende og funksjonalitet` under Site Settings.

Here's a sample .txt file:

```text
{
    "ComposedLook": {
        "ColorPaletteUrl": "{sitecollection}/_catalogs/theme/15/MyColors.spcolor",
        "FontSchemeUrl": "{sitecollection}/_catalogs/theme/15/MyFonts.spfont",
        "BackgroundImageUrl": null
    }
}
```

**Comment by PrinsMild** [comment]
Good. That worked fine. But the logo still remains PP default and not the one chosen on the root site.

**Comment by olemp** [comment]
If you want another logo you need to override``WebSettings`.

Here's a sample .txt file:

```text
{
    "WebSettings": {
        "SiteLogoUrl": "{sitecollection}/SiteAssets/YourLogo.png"
    }
}

**Comment by olemp** [comment]
You can combine those two if you'd like:

```text
{
    "WebSettings": {
        "SiteLogoUrl": "{sitecollection}/SiteAssets/YourLogo.png"
    },
 "ComposedLook": {
        "ColorPaletteUrl": "{sitecollection}/_catalogs/theme/15/MyColors.spcolor",
        "FontSchemeUrl": "{sitecollection}/_catalogs/theme/15/MyFonts.spfont",
        "BackgroundImageUrl": null
    }
}
```

**Comment by PrinsMild** [comment]
When updating the portal to version 2.1.16 after adding the settings suggested, I've got an error message:
"Cannot bind parameter 'ForegroundColor' to the target. Exception setting "ForegroundColor": "Object reference not set to an instance of an object." At Upgrade.ps1 line 85 char 5. 

Line 85 is calling Install.ps1

When upgrading another installation without these settings, no error ocurrs.

**Comment by olemp** [comment]
@PrinsMild I'll have a look at this asap. It's not related to `Selectable theme for projects (sub sites)` so create a new issue next time and it'll be easier for us to find it. 👍

**Comment by olemp** [comment]
Should be solved for `2.1.17` @PrinsMild: https://github.com/Puzzlepart/prosjektportalen/releases/tag/v2.1.17
